shell/platform/darwin/macos/framework/Source/FLEExternalTextureGL.mm

```diff
@@ -0,0 +1,86 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/macos/framework/Source/FLEExternalTextureGL.h"
+
+#import <AppKit/NSOpenGL.h>
+#import <CoreVideo/CVOpenGLBuffer.h>
+#import <CoreVideo/CVOpenGLTextureCache.h>
+#import <OpenGL/gl.h>
+
+@implementation FLEExternalTextureGL {
+  CVOpenGLTextureCacheRef _textureCache;
+  CVPixelBufferRef _pixelBuffer;
+  id<FLETexture> _fleTexture;
+}
+
+- (instancetype)initWithFLETexture:(id<FLETexture>)fleTexture {
+  self = [super init];
+
+  if (self) {
+    _fleTexture = fleTexture;
+  }
+
+  return self;
+}
+
+static void OnGLTextureRelease(CVPixelBufferRef pixelBuffer) {
+  CVPixelBufferRelease(pixelBuffer);
+}
+
+- (int64_t)textureId {
+  return (NSInteger)(self);
+}
+
+- (BOOL)populateTextureWidth:(size_t)width
+                      height:(size_t)height
+                     texture:(FlutterOpenGLTexture*)texture {
+  if (_fleTexture == NULL) {
```

Comment:
The preferred style for the macOS shell is:
if (!_fleTexture)

Same with the other pointer comparisons below.